### PR TITLE
Fix Notice Error: Trying to access array offset on value of type null

### PR DESCRIPTION
**Pull request by ravage84**
On PHP 7.4 when providing ``null`` for the ``$path``

### CODE REVIEW

**Comment by ADmad** [comment]
A null path doesn't make sense, I would rather throw an exception.

**Comment by dereuromark** [comment]
Or we cast :) Thats safer

**Comment by ADmad** [comment]
It's not safer, it's masking an invalid argument.

**Comment by dereuromark** [comment]
It might not be documented, but so far it worked. Breaking it for no reason is also not reasable IMO.

**Comment by ravage84** [comment]
> A null path doesn't make sense, I would rather throw an exception.

@ADmad  while I agree that null is actually an invalid argument, throwing an exception would blow up the whole website. A bit extreme...

**Comment by dereuromark** [comment]
@ravage84 Do you have an example where this would happen and make sense to have null and/or empty string supported?

**Comment by ravage84** [comment]
> 
> 
> @ravage84 Do you have an example where this would happen and make sense to have null and/or empty string supported?

@dereuromark to be honest, I found this case in a data provider set in one of our test cases. Not sure, it was deliberately or one of our junior devs wanted to be thorough, as I thaught them to be. :laughing:

**Comment by ADmad** [comment]
Ah this is for 3.x, well then ignore my comments, do whatever hack is necessary to avoid the error :)

**Review by ADmad — APPROVED** [review]

File: src/View/Helper/UrlHelper.php [hunk 1]
```diff
@@ -174,7 +174,7 @@ public function assetUrl($path, array $options = [])
         if (!array_key_exists('plugin', $options) || $options['plugin'] !== false) {
             list($plugin, $path) = $this->_View->pluginSplit($path, false);
         }
-        if (!empty($options['pathPrefix']) && $path[0] !== '/') {
+        if (!empty($options['pathPrefix']) && ($path === null || $path[0] !== '/')) {
```
Comment (dereuromark) [inline src/View/Helper/UrlHelper.php]: With an empty string you have the same issue...
Why not using `substr((string)...` here, that is safer.

**Pull request merged**